### PR TITLE
Default statistics backward compatible change

### PR DESCRIPTION
**Pull request by rcillo**
In the PR where we changed the repartitioning endpoint we also
accidentally introduced a change in an undocumented behaviour that
some clients relied on. It used to be possible to update an event type
without providing default statistics.

We believe this behaviour should not be modified because this field is
optional, so some clients might not even support it (Nakadi UI, for
example).

We are also fixing a problem where repartition could be triggered
concurrently with an event type update resulting in some race
condition on database and timeline changes in database.

### CODE REVIEW

File: core-services/src/main/java/org/zalando/nakadi/service/RepartitioningService.java [hunk 1]
```diff
@@ -89,18 +99,44 @@ public void repartition(final EventType eventType, final int partitions)
...
+            Thread.currentThread().interrupt();
+            LOG.error("Failed to wait for timeline switch", e);
+            throw new EventTypeUnavailableException("Event type " + eventType.getName()
+                    + " is currently in maintenance, please repeat request");
+        } catch (final TimeoutException e) {
+            LOG.error("Failed to wait for timeline switch", e);
+            throw new EventTypeUnavailableException("Event type " + eventType.getName()
+                    + " is currently in maintenance, please repeat request");
+        } catch (final InternalNakadiException | ServiceTemporarilyUnavailableException e) {
+            LOG.error("Error deleting event type " + eventType.getName(), e);
```
Comment (adyach) [inline core-services/src/main/java/org/zalando/nakadi/service/RepartitioningService.java:130]: do we need to process this execeptions here? if so could you adjust the message please?

Comment (rcillo) [reply core-services/src/main/java/org/zalando/nakadi/service/RepartitioningService.java]: fixed 7b017fe0a

**Comment by adyach** [comment]
👍

**Comment by rcillo** [comment]
:+1:

**Comment by adyach** [comment]
👍

**Pull request merged**